presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -365,6 +366,8 @@ public IPrestoSparkQueryExecution create(
         log.info("Starting execution for presto query: %s", queryId);
         System.out.printf("Query id: %s\n", queryId);
 
+        sparkContext.conf().set(PRESTO_QUERY_ID_CONFIG, queryId.getId());
```

Comment:
What is very weird is this isn't in the javadoc! I assumed you could only get a read only copy.

Reply:
if you do getConf() - you get clone of config which doesn't reflect back updates. This method call returns underlying conf object.

Reply:
Yeah I am just surprised that `conf()` isn't in the [javadoc](https://spark.apache.org/docs/2.4.6/api/java/index.html).

Reply:
That could be because its marked package private from scala perspective ?